File: library/src/main/java/com/jdroid/java/exception/UnexpectedException.java
package com.jdroid.java.exception;


public class UnexpectedException extends ErrorCodeException {
	
	private static final long serialVersionUID = 7653653326088130933L;
	
	private Boolean useCause = false;
	
	public UnexpectedException(String message, Throwable cause) {
		super(CommonErrorCode.UNEXPECTED_ERROR, message, cause);
		setTrackable(true);
	}
	
	public UnexpectedException(String message) {
		super(CommonErrorCode.UNEXPECTED_ERROR, message);
		setTrackable(true);
	}
	
	public UnexpectedException(Throwable cause) {
		super(CommonErrorCode.UNEXPECTED_ERROR, cause);
		setTrackable(true);
		useCause = true;
	}
	
	@Override
	public Throwable getThrowableToLog() {
		return (getCause() != null) && useCause ? getCause() : super.getThrowableToLog();
	}
}

File: jdroid-java-core/src/main/java/com/jdroid/java/exception/UnexpectedException.kt
package com.jdroid.java.exception

class UnexpectedException : ErrorCodeException {

    private var useCause = false

    constructor(message: String, cause: Throwable) : super(CommonErrorCode.UNEXPECTED_ERROR, message, cause) {
        isTrackable = true
    }

    constructor(message: String) : super(CommonErrorCode.UNEXPECTED_ERROR, message) {
        isTrackable = true
    }

    constructor(cause: Throwable) : super(CommonErrorCode.UNEXPECTED_ERROR, cause) {
        isTrackable = true
        useCause = true
    }

    override fun getThrowableToLog(): Throwable {
        return if (cause != null && useCause) cause else super.getThrowableToLog()
    }
}